maestrowf/datastructures/yamlspecification.py

```diff
@@ -41,6 +41,42 @@
 
 logger = logging.getLogger(__name__)
 
+STUDY_STEP_KEYS = {
```

Comment:
I think that these might be better in the `YAMLSpecification` class itself. There's another idea I had thinking about this PR, but it might be overkill in terms of requiring a file read and such. You could make a fixed reference file in JSON/YAML (JSON is probably better because it serializes much faster). There's a MANIFEST file that can be added to a Python package to denote fixed data files.

The other option is to explore the jobspec package that @kcathey pointed to, if you'd be willing to take a look. Might be use to work with.

Reply:
Sure, I'll look at the package first. In the meantime, can you fill me in on the keys I'm missing?

Reply:
For the steps, this should be the set https://github.com/LLNL/maestrowf/blob/develop/maestrowf/datastructures/core/study.py#L67

For the `env` section:
- `variable` section keys are simply `name: key`
- `labels` section is the same as variables, except you can use variables and parameters.
- `dependencies` - You're right except there are more possible keys in the `git` dependency section. Those are not required though.

Reply:
Is there a set of all the allowed keys in the git dependency section?

Reply:
The keys are unofficially documented in the code here:
https://github.com/LLNL/maestrowf/blob/develop/maestrowf/datastructures/environment/gitdependency.py#L122